custom_components/waste_collection_schedule/waste_collection_schedule/source/aw_harburg.py

```diff
@@ -0,0 +1,203 @@
+from waste_collection_schedule.service.ICS import ICS
+from bs4 import BeautifulSoup
+
+TITLE = "AW Harburg"
+DESCRIPTION = "Abfallwirtschaft Landkreis Harburg"
+URL = "https://www.landkreis-harburg.de/bauen-umwelt/abfallwirtschaft/abfallkalender/"
+
+TEST_CASES = {
+    "CityWithTwoLevels": {
+        "district_level_1": "Hanstedt",
+        "district_level_2": "Evendorf",
+    },
+    "CityWithThreeLevels": {
+        "district_level_1": "Buchholz",
+        "district_level_2": "Buchholz mit Steinbeck (ohne Reindorf)",
+        "district_level_3": "Seppenser Mühlenweg Haus-Nr. 1 / 2",
+    },
+}
+
+class Source:
+    def __init__(self, district_level_1, district_level_2, district_level_3=None):
+        self._district_level_1 = district_level_1
+        self._district_level_2 = district_level_2
+        self._district_level_3 = district_level_3
+        self._ics = ICS()
+
+    def fetch(self):
+        # Use a session to keep cookies and stuff
+        s = requests.Session()
+        
+        # Creat some fake header because for some reason people seem to believe it is bad
+        # to read public garbage collection data via a script
+        headers = {
+            'Connection': 'keep-alive',
+            'Cache-Control': 'max-age=0',
+            'sec-ch-ua': '" Not A;Brand";v="99", "Chromium";v="98", "Opera";v="84"',
+            'sec-ch-ua-mobile': '?0',
+            'sec-ch-ua-platform': '"Windows"',
+            'Upgrade-Insecure-Requests': '1',
+            'Content-Type': 'application/x-www-form-urlencoded',
+            'User-Agent': 'Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/98.0.4758.102 Safari/537.36 OPR/84.0.4316.21',
+            'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3;q=0.9',
+            'Sec-Fetch-Site': 'same-origin',
+            'Sec-Fetch-Mode': 'navigate',
+            'Sec-Fetch-User': '?1',
+            'Sec-Fetch-Dest': 'document',
+            'Accept-Language': 'de-DE,de;q=0.9',
+        }
+
+        # Get the IDs of the districts on the first level
+        # Double loading is on purpose because sometimes the webpage has an overlay
+        # which is gone on the second try in a session
+        response = s.get(URL, headers=headers)
+        if "Zur aufgerufenen Seite" in response.text:
+            response = s.get(URL, headers=headers)
+        if response.status_code != 200:
+            raise Exception(
+                "Error: failed to fetch first url: {}".format(
+                    URL
+                )
+            )
+        soup = BeautifulSoup(response.text, features="html.parser")
+        select_content = soup.find_all("select", id="strukturEbene1")
+        soup = BeautifulSoup(str(select_content), features="html.parser")
+        options_content = soup.find_all("option")
+        level_1_ids = {}
+        for option in options_content:
```

Comment:
The parsing of the 3 levels looks like a lot of code duplication. Could this be separated into a function?